python/cudaq/kernel/ast_bridge.py

```diff
@@ -402,6 +406,13 @@ def promoteOperandType(self, ty, operand):
             if IntegerType.isinstance(operand.type):
                 operand = arith.SIToFPOp(ty, operand).result
 
+        if IntegerType.isinstance(ty):
+            if IntegerType.isinstance(operand.type):
+                if IntegerType(ty).width < IntegerType(operand.type).width:
+                    operand = arith.TruncIOp(ty, operand).result
+                else:
+                    operand = arith.ExtSIOp(ty, operand).result
```

Comment:
Shouldn't these just use cc.cast here?

Reply:
yes, I am planning to send a separate PR for this one and many other cases.